src/App.tsx

```diff
@@ -1,24 +1,136 @@
-/* eslint-disable max-len */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
-import { UserWarning } from './UserWarning';
+import React, { useEffect, useState } from 'react';
+import { createTodo, deleteTodo, getTodos } from './api/todos';
+import { Header } from './Components/Header';
+import { Main } from './Components/Main';
+import { Footer } from './Components/Footer';
+import { Todo } from './types/Todo';
+import { Filter } from './types/Filter';
+import { Errors } from './types/Errors';
 
-const USER_ID = 0;
+const USER_ID = 10307;
 
 export const App: React.FC = () => {
-  if (!USER_ID) {
-    return <UserWarning />;
-  }
+  // if (!USER_ID) {
+  //   return <UserWarning />;
+  // }
+
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [errors, setErrors] = useState<Errors | null>(null);
+  const [filter, setFilter] = useState<Filter>(Filter.All);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [isDeleting, setIsDeleting] = useState<number[]>([]);
+
+  const addTodo = (todoTitle: string) => {
+    const todo: Todo = {
+      title: todoTitle,
+      userId: USER_ID,
+      completed: false,
+    };
+
+    setTempTodo({
+      id: 0,
+      title: todoTitle,
+      userId: USER_ID,
+      completed: false,
+    });
+
+    createTodo(todo)
+      .then((value) => {
+        setTodos([...todos, value]);
+        setTempTodo(null);
+      })
+      .catch(() => {
+        setErrors({ posting: true });
+        setTempTodo(null);
+      });
+  };
+
+  const removeTodo = (id: number) => {
```

Comment:
would be better to do it with `async` function
also, wrap it with `useCallback`